tests/test_jobqueue.py

```diff
@@ -68,21 +66,43 @@ def test_init_job(self):
 @pytest.mark.skipif(
     not TEST_WITH_OPT_DEPS, reason="Only relevant if the optional dependency is installed"
 )
-@pytest.mark.skipif(
-    os.getenv("GITHUB_ACTIONS", False) and platform.system() in ["Windows", "Darwin"],
-    reason="On Windows & MacOS precise timings are not accurate.",
-)
-@pytest.mark.flaky(10, 1)  # Timings aren't quite perfect
+# needed if we run pytest -n=number > no. of CPUs & since asyncio.Events aren't thread-safe.
+@pytest.mark.flaky(4, 1)
 class TestJobQueue:
     result = 0
     job_time = 0
     received_error = None
+    # we have 2 more variables which define in `reset` since it needs to be on the same event loop
 
     @pytest.fixture(autouse=True)
     def reset(self):
         self.result = 0
         self.job_time = 0
         self.received_error = None
+        self.done_with_cb = asyncio.Event()
+        self.done_with_err_cb = asyncio.Event()
+
+    async def wait_for_cbs(self, number: int = 1):
+        async def _wait_for_cbs():
+            for _ in range(number):
+                await self.done_with_cb.wait()
+                self.done_with_cb.clear()
```

Comment:
maybe `asyncio.Barrier` can help here?

Reply:
I would use that, but it's py 3.11+ only..

Reply:
ups, overlooked that …